src/app.js

```diff
@@ -1,342 +1,232 @@
+   * Sets the app lifecycle to not running.
+   * Removes any listeners added during the running state
    *
    * @public
-   * @method getChildApps
+   * @method stop
    * @memberOf App
-   * @returns {Array}
+   * @param {Object} [options] - Settings for the App passed through to events
+   * @event AbstractApp#before:stop - passes options
+   * @event AbstractApp#stop - passes options
+   * @returns {AbstractApp}
    */
-  getChildApps: function() {
-    return _.clone(this._childApps);
-  },
+  stop(options) {
+    if(!this._isRunning) {
+      return this;
+    }
 
-  /**
-   * Returns registered child `App`
-   *
-   * @public
-   * @method getChildApp
-   * @memberOf App
-   * @param {String} appName - Name of App to retrieve
-   * @returns {App}
-   */
-  getChildApp: function(appName) {
-    return this._childApps[appName];
-  },
+    this.triggerMethod('before:stop', options);
 
-  /**
-   * Internal helper.  Unregisters child `App`
-   *
-   * @private
-   * @method _removeChildApp
-   * @memberOf App
-   * @param {String} appName - Name of App to unregister
-   * @returns {App}
-   */
-  _removeChildApp: function(appName) {
-    delete this._childApps[appName]._name;
-    delete this._childApps[appName];
+    this._isRunning = false;
+
+    this.triggerMethod('stop', options);
+
+    this._stopRunningListeners();
+    this._stopRunningEvents();
+
+    return this;
   },
 
   /**
-   * Removes all childApps and returns them.
-   * The return is useful if any app is using `preventDestroy`
+   * Gets the value of internal `_isDestroyed` flag
    *
    * @public
-   * @method removeChildApps
+   * @method isDestroyed
    * @memberOf App
-   * @returns {Array}
+   * @returns {Boolean}
    */
-  removeChildApps: function() {
-    const childApps = this.getChildApps();
-
-    _.each(this._childApps, function(childApp, appName) {
-      this.removeChildApp(appName);
-    }, this);
-
-    return childApps;
+  isDestroyed() {
+    return this._isDestroyed;
   },
 
   /**
-   * Destroys or removes registered child `App` by name
-   * depending on `preventDestroy`
+   * Stops the `App` and sets it destroyed.
    *
    * @public
-   * @method removeChildApp
+   * @method destroy
    * @memberOf App
-   * @param {String} appName - Name of App to destroy
-   * @param {Object} [options.preventDestroy] - Flag to remove but prevent App destroy
-   * @returns {App}
    */
-  removeChildApp: function(appName, options) {
-    options = _.extend({}, options);
-
-    const childApp = this.getChildApp(appName);
-
-    if(!childApp) {
+  destroy() {
+    if(this._isDestroyed) {
       return;
     }
 
-    // if preventDestroy simply unregister the child app
-    if(options.preventDestroy || _.result(childApp, 'preventDestroy')) {
-      this._removeChildApp(appName);
-    } else {
-      childApp.destroy();
-    }
+    this._isDestroyed = true;
 
-    return childApp;
-  }
+    this.stop();
+    this.destroyState();
```

Comment:
`destroyState` should be privatized and in `initState` we should `this.on('destroy', this._destroyState);`  I don't know how that changes `initState`'s reusability, but in order to install the mixin we shouldn't require 2 functions